### PR TITLE
Transport error module

### PR DESCRIPTION
**Pull request by vaporos**
Move errors in transport:: to transport::errors and implement Error and Display for all errors.

### CODE REVIEW

File: libsplinter/src/transport/error.rs [hunk 1]
```diff
@@ -21,6 +21,24 @@ pub enum AcceptError {
     ProtocolError(String),
 }
 
+impl Error for AcceptError {
```
Comment (agunde406) [inline libsplinter/src/transport/error.rs]: Should "ghe above " -> Say "the above"?

Comment (agunde406) [reply libsplinter/src/transport/error.rs]: In the commit message

Comment (ltseeley) [reply libsplinter/src/transport/error.rs]: Also s/is common/is a common/

Comment (vaporos) [reply libsplinter/src/transport/error.rs]: fixed

File: libsplinter/src/transport/mod.rs [hunk 2]
```diff
@@ -22,10 +23,12 @@ pub mod tls;
 #[cfg(feature = "zmq-transport")]
 pub mod zmq;
 
-use mio::Evented;
+pub use error::{
+    AcceptError, ConnectError, DisconnectError, ListenError, PollError, RecvError, SendError,
+    StatusError,
+};
```
[the remark this answers is not in the document]
Comment (vaporos) [reply libsplinter/src/transport/mod.rs]: fixed

File: libsplinter/src/transport/mod.rs [hunk 3]
```diff
@@ -24,13 +25,14 @@ pub mod zmq;
...
-use std::io;
+pub use error::{
+    AcceptError, ConnectError, DisconnectError, ListenError, PollError, RecvError, SendError,
+    StatusError,
+};
 
 pub enum Status {
     Connected,
     Disconnected,
-}
```
Comment (peterschwarz) [inline libsplinter/src/transport/mod.rs]: I can't tell if this is a github problem or an accidental deletion.

Comment (peterschwarz) [reply libsplinter/src/transport/mod.rs]: From the looks of it, the rest of the enum is deleted in a later commit

**Pull request merged**